### PR TITLE
Release/1.3.2

### PR DESCRIPTION
**Pull request by chuwy**

### CODE REVIEW

**Review by dilyand — APPROVED** [review]
Looks great @chuwy. I think there's just one typo that needs fixing before release (see comment).

File: modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/ConversionUtils.scala [hunk 1]
```diff
@@ -471,4 +473,8 @@ object ConversionUtils {
       nvps = r.asScala.toList
       pairs = nvps.map(p => p.getName() -> p.getValue())
     } yield pairs.toMap
+
+  /** Extract valid IP (v4 or v6) address from a string */
+  def extractInetAddress(arg: String): Option[InetAddress] =
```
Comment (benjben) [inline modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/ConversionUtils.scala:478]: Would be great to add a unit test

File: modules/common/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/EnrichmentManagerSpec.scala [hunk 2]
```diff
@@ -321,4 +306,96 @@ class EnrichmentManagerSpec extends Specification with EitherMatchers {
       enriched.value.map(_.useragent) must beRight("header-useragent")
     }
   }
+
+  "getIabContext" should {
```
Comment (benjben) [inline modules/common/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/EnrichmentManagerSpec.scala:310]: Awesome that you added great testing!

File: modules/beam/src/test/scala/com.snowplowanalytics.snowplow.enrich.beam/enrichments/IpLookupsEnrichmentSpec.scala [hunk 3]
```diff
@@ -36,12 +36,12 @@ object IpLookupsEnrichmentSpec {
...
-    "A91",
-    "53.999",
-    "-6.4183",
-    "Louth",
+    "L",
+    "Dublin",
+    "D02",
+    "53.3331",
+    "-6.2489",
+    "Leinster",
```
Comment (benjben) [inline modules/beam/src/test/scala/com.snowplowanalytics.snowplow.enrich.beam/enrichments/IpLookupsEnrichmentSpec.scala:44]: We should really take care of https://github.com/snowplow/enrich/issues/269 at some point

**Review by benjben — APPROVED** [review]
It looks great @chuwy ! IAB enrichment is so much cleaner as before!

File: modules/common/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/registry/IabEnrichmentSpec.scala [hunk 4]
```diff
@@ -92,24 +91,21 @@ class IabEnrichmentSpec extends Specification with DataTables {
...
 
-  def e2 =
-    validConfig.enrichment[Eval].map(_.performCheck("", "foo//bar", DateTime.now())).value must
-      beLeft
-
-  def e3 =
-    validConfig.enrichment[Eval].map(_.getIabContext(None, None, None)).value must beLeft
-
-  def e4 = {
+  def e2 = {
```
[the remark this answers is not in the document]
Comment (chuwy) [reply modules/common/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/registry/IabEnrichmentSpec.scala:94]: Yup, reason is that types do not match anymore. Previously we were testing `null`s as inputs and they're not possible in there anymore.

File: modules/common/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/EnrichmentManagerSpec.scala [hunk 5]
```diff
@@ -321,4 +306,96 @@ class EnrichmentManagerSpec extends Specification with EitherMatchers {
...
+
+    "return None if user_ipaddress in invalid" >> {
+      val input = new EnrichedEvent()
+      input.setUser_ipaddress("invalid")
+      input.setUseragent("Firefox")
+      input.setDerived_tstamp("2010-06-30 01:20:01.000")
+      EnrichmentManager.getIabContext(input, iabEnrichment) must beRight(None)
+    }
+
+    "return Some if all arguments are valid invalid" >> {
```
[the remark this answers is not in the document]
Comment (chuwy) [reply modules/common/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/EnrichmentManagerSpec.scala:340]: Certainly.

File: modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/ConversionUtils.scala (hunk 1 above)
Comment (chuwy) [reply modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/ConversionUtils.scala:478]: Yup, you're right, coveralls notified about dropped coverage and think this is the reason.

**Pull request merged**